controllers/terraform_controller.go

```diff
@@ -623,6 +624,18 @@ func (r *TerraformReconciler) plan(ctx context.Context, terraform infrav1.Terraf
 	planRev := strings.Replace(revision, "/", "-", 1)
 	planName := "plan-" + planRev
 
+	if _, hasEncoding := terraform.ObjectMeta.Annotations["encoding"]; hasEncoding {
+		tfplan, err = r.encodePlan(terraform, tfplan)
+		if err != nil {
+			return infrav1.TerraformNotReady(
+				terraform,
+				revision,
+				infrav1.TFExecPlanFailedReason,
+				err.Error(),
+			), err
+		}
+	}
+
 	tfplanData := map[string][]byte{"tfplan": tfplan}
 	tfplanSecret = corev1.Secret{
 		TypeMeta: metav1.TypeMeta{
```

Comment:
Please set `metadata: annotations: encoding: "gzip"` here for the tfplan secret.